source/code/plugins/in_npmd_server.rb

```diff
@@ -82,52 +46,116 @@ def configure(conf)
         EXIT_RESTART_BACKOFF_THRES_SECS = 900
 
         NPMD_AGENT_CAPABILITY = "cap_net_raw+ep"
-        NPMD_VERSION_FILE_DIR = "state"
+        NPMD_STATE_DIR = "/var/opt/microsoft/omsagent/npm_state"
         NPMD_VERSION_FILE_NAME = "npm_version"
 
+        config_param :omsadmin_conf_path,     :string, :default => "/etc/opt/microsoft/omsagent/conf/omsadmin.conf"
+        config_param :location_unix_endpoint, :string, :default => "#{NPMD_STATE_DIR}/npmdagent.sock"
+        config_param :location_control_data,  :string, :default => "/etc/opt/microsoft/omagent/conf/npmd_agent_config.xml"
+        config_param :location_agent_binary,  :string, :default => "/opt/microsoft/omsagent/plugin/npmd_agent"
+        config_param :tag, :string, :default => "oms.npmd"
+
+        def configure(conf)
+            super
+
+            unless @omsadmin_conf_path
```

Comment:
I'm not sure that this `unless` clause is necessary, since you did provide a default value for `omsadmin_conf_path` above.